src/main/java/moonset/metastore/sync/MetastoreSyncUtils.java

```diff
@@ -0,0 +1,200 @@
+                log.info(String.format("The table original properties: %s", destTable.getParameters()));
+                Map<String, String> parameters = destTable.getParameters();
+                parameters.putAll(source.getTable(srcDatabaseName, srcTableName).getParameters());
+                log.info(String.format("The table properties after updated: %s", parameters));
+                destTable.setParameters(parameters);
+                dest.alter_table(destDatabaseName, destTableName, destTable);
+                log.info("The table properties updated successfully.");
+            } else {
+                log.info("The table " + destTableName + " doesn't exist in destination metastore, and start to sync it.");
+                Table sourceTable = source.getTable(srcDatabaseName, srcTableName);
+                log.info("Rewrite the table EXTERNAL paramater, which is used by Hive Metastore.");
+                Table modifiedTable = TableUtils.rewriteExternalFlag(sourceTable);
+                // Rename table name to destTableName.
+                modifiedTable.setDbName(destDatabaseName);
+                modifiedTable.setTableName(destTableName);
+
+                //Store the create time in table parameter, sice this field will be overriden when createTable().
+                if(!modifiedTable.getParameters().containsKey(TableUtils.ORIGINAL_CREATE_TIME)) {
+                    modifiedTable.getParameters().put(TableUtils.ORIGINAL_CREATE_TIME, String.valueOf(modifiedTable.getCreateTime()));
+                }
+                dest.createTable(modifiedTable);
+                log.trace("The table: " + modifiedTable);
+                log.info("The table " + destTableName + " is created successfully.");
+            }
+
+            //validate dest table
+            if (!"TRUE".equals(dest.getTable(destDatabaseName, destTableName).getParameters().get(TableUtils.EXTERNAL_PARAM))) {
+                throw new MetastoreException(
+                        "It's dangerous to process a non-external table. Prohibit to do so.");
+            }
+        } catch (TException e) {
+            throw new MetastoreException("failed to sync table", e);
+        }
+    }
+
+    /**
+     * The partitions should be processed in small batch, since the server creates a thread for each
+     * partition and make a transaction for each call. Without traffic control, too much threads
+     * will be created and overwhelm the server.
+     */
+    public static void syncPartitions(
+            final IMetaStoreClient dest,
+            final String destDatabaseName,
+            final String destTableName,
+            final List<Partition> partitions)
+            throws MetastoreException {
+        try {
+            if (CollectionUtils.isEmpty(partitions)) {
+                // We only log empty partitions but do nothing
+                log.info("No partition found, skip sync partitions.");
+            } else {
```

Comment:
If we do nothing for empty partitions, this else branch is unnecessary
```suggestion
            if (CollectionUtils.isEmpty(partitions)) {
                // We only log empty partitions but do nothing
                log.info("No partition found, skip sync partitions.");
            }
            log.info("....
            ... ...
            ... ...
```